pkg/operator/ceph/cluster/cluster.go

```diff
@@ -139,46 +140,60 @@ func (c *cluster) createInstance(rookImage string) error {
 
 	_, err := c.context.Clientset.CoreV1().ConfigMaps(c.Namespace).Create(cm)
 	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("failed to create override configmap %s. %+v", c.Namespace, err)
+		errMsg := "failed to create override configmap %s. %+v"
+		c.status.SetMessage(fmt.Sprintf(errMsg, c.Namespace, err))
```

Comment:
Could you call `status.SetMessage()` from the caller of `createInstance()`? Then we don't have to call it in so many places here.

Reply:
I have added it here so that we can capture the fine grained failure message. 
Should I move it outside?

Reply:
Yes, it will be nice if we only have to call it once instead of so many different places when there is an error.